test/harness/e2e/harness.go

```
@@ -75,60 +75,105 @@ func NewTestHarness() *Harness {
 	ctx, cancel := context.WithCancel(context.Background())
 
 	return &Harness{
-		VM:        testVM,
+		VMs:       []vm.TestVMInterface{testVM},
 		Client:    c,
 		Context:   ctx,
 		ctxCancel: cancel,
 		startTime: startTime,
 	}
 }
 
+func (h *Harness) DefaultVM() vm.TestVMInterface {
+    if len(h.VMs) == 0 {
+        Fail("No VMs available in the harness")
+    }
+    return h.VMs[0]
+}
+
+func (h *Harness) VM() vm.TestVMInterface {
+    return h.DefaultVM()
+}
+
+func (h *Harness) AddVM(vmParams vm.TestVM) vm.TestVMInterface {
+    testVM, err := vm.NewVM(vmParams)
+    Expect(err).ToNot(HaveOccurred())
```

We are trying to use Expect only in the tests, since it is possible that in validations tests we want to see an error. So maybe here you could return an error, and do:
```
if err != nil {
  return nil, err
}